### ISSUE TITLE
Tabs

### ISSUE THREAD

**Issue by isAlmogK** [angular-ui/angular-ui-OLDREPO#212]
I don't see an option for tabs, is this something you plan on providing? Is there way to use Bootstrap tabs with Angular UI?

**Comment by ProLoser** [comment]
There is no explicit solution for tabs. You can easily recreate the functionality by using ng-classes. We have been discussing creating a playground of demos on how to use any functionality not explicitly covered.

The benefit you gain of doing it manually is that you can then programmatically and easily change the active tabs from anywhere.

**Comment by isAlmogK** [comment]
Thanks for the quick reply was just talking on Google Groups, if you have any examples you can share please let me know. 

Bye the way Angular UI is great.

**Comment by ProLoser** [comment]
I copy-pasted the source from the demo page and it works perfectly fine :)

No angular-ization necessary :)

**Comment by isAlmogK** [comment]
Yea I understand I guess my user case is a bit more complex. I need each tab to work with a route and load a different view. Not to mention I'm still not sure which view the tabs need to be under my main index.html file has ng-view and just loads different views however  each view is a tab.

**Comment by ProLoser** [comment]
That would be exactly why we don't have our own implementation.

Do what I do: use the classes and ng-class. If you inspect the source code you should be able to easily see what classes are added or removed. The only caveat is that you cannot do animations alongside routing (or at least, it's very complex).

I use tabs and routes in our app, but we just use simple stuff like `<ul class="tabs"><li ng-class="{active:currUrl()}">` or something of that nature

**Comment by isAlmogK** [comment]
Yea tell me about it, don't know why really stuck on this. I think I might fake it and create a menu which looks like tabs. But will check out class's thanks for all the help.

**Comment by ProLoser** [comment]
Again nothing is stopping you from leveraging Bootstrap's classes to get
the same visual layout and behavior.

On Sat, Sep 29, 2012 at 10:38 PM, Almog Koren notifications@github.comwrote:

> Yea tell me about it, don't know why really stuck on this. I think I might
> fake it and create a menu which looks like tabs. But will check out class's
> thanks for all the help.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/angular-ui/angular-ui/issues/212#issuecomment-9011081.

## 

Dean J Sofer
DeanSofer@gmail.com

BS Computer Information Systems
California State Polytechnic University, Pomona
Telephone: 714-900-2254
Website: www.DeanSofer.com
Github: github.com/ProLoser

**Comment by isAlmogK** [comment]
Thanks it's a little beyond me right now, still learning angular going pretty smooth or ok so far only issue is the tabs and I'm just going to do it like our current version which is more like menus. You can look at what I'm working on scoreoid.com the current version is CakePHP I'm working on the new version which is a pure webapp with rails backend.

**Comment by ProLoser** [comment]
Viewing the source code to see the bootstrap classes has nothing to do with
angular.

On Sat, Sep 29, 2012 at 11:05 PM, Almog Koren notifications@github.comwrote:

> Thanks it's a little beyond me right now, still learning angular going
> pretty smooth or ok so far only issue is the tabs and I'm just going to do
> it like our current version which is more like menus. You can look at what
> I'm working on scoreoid.com the current version is CakePHP I'm working on
> the new version which is a pure webapp with rails backend.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/angular-ui/angular-ui/issues/212#issuecomment-9011190.

## 

Dean J Sofer
DeanSofer@gmail.com

BS Computer Information Systems
California State Polytechnic University, Pomona
Telephone: 714-900-2254
Website: www.DeanSofer.com
Github: github.com/ProLoser

**Comment by isAlmogK** [comment]
Yes I know that, my issue wasn't with bootstrap but that correct way to load the tab content which is angular view with it's one model, controller. And I can't do it on the main index.html file.

**Comment by ajoslin** [comment]
Hi @almogdesign, here's an example of bootstrap tabs without a directive that I did awhile ago: https://gist.github.com/3181630

It's pretty simple, which is why we don't think a directive is needed

(I didn't test it, but I'm almost sure it will work :-)

**Comment by ProLoser** [comment]
View my comments on Andy's gist to see this implemented with proper routes:
https://gist.github.com/3813208#comments

On Mon, Oct 1, 2012 at 6:50 AM, Andy Joslin notifications@github.comwrote:

> Hi almog, here's an example of tabs I did awhile ago:
> https://gist.github.com/3181630
> 
> (I didn't test it, but I'm almost sure it will work :-)
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/angular-ui/angular-ui/issues/212#issuecomment-9032509.

## 

Dean J Sofer
DeanSofer@gmail.com

BS Computer Information Systems
California State Polytechnic University, Pomona
Telephone: 714-900-2254
Website: www.DeanSofer.com
Github: github.com/ProLoser

**Comment by pkozlowski-opensource** [comment]
We've got now bootstrap tabs directive in a separate repo (https://github.com/angular-ui/bootstrap/blob/master/src/tabs/tabs.js) that works fine for typical uses cases.

If the above directive doesn't fit your needs it is relatively simple to crate a custom implementation (as pointed out by @ajoslin  and @ProLoser).

Closing this issue for now, open a separate issue in the boostrap repo if you believe that things are missing.
